### PR TITLE
Fix vat for percent promotions

### LINKED ISSUE #6548: Cannot delete line_item through api with address
Discovered this in sandbox 3-0-stable but presumably it would exist in master as well. The recent commit of #6542 makes the TaxRate be recreated after adjustments are made which seems to cause issues when attempting to delete a line_item through the api with address.

Two things:
* I am not sure [this](https://github.com/spree/spree/blob/master/core/app/models/spree/order_contents.rb#L43) line is necessary, because why are we creating tax before we even have a location? Any details @vrosancic?

* When a line_item is destroyed TaxRate attempts to [update_column](https://github.com/spree/spree/blob/master/core/app/models/spree/tax_rate.rb#L47) on the destroyed line_item raising a ```{ "exception": "cannot update a destroyed record" }```

Steps to replicate: 
1. add line_item to cart
2. proceed to checkout entering a valid address
3. DELETE request to api/orders/:order_number/line_items/:line_item_id with a valid token

### PR DESCRIPTION
**Pull request by vrosancic**
Fixes #6548

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://vrosancic-fix-vat-for-percent-promotions.spree-spree-9ed724b77adec04c49c2.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://vrosancic-fix-vat-for-percent-promotions.spree-spree-9ed724b77adec04c49c2.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

File: core/app/models/spree/order_contents.rb [hunk 1]
```diff
@@ -40,6 +40,7 @@ def after_add_or_remove(line_item, options = {})
       shipment.present? ? shipment.update_amounts : order.ensure_updated_shipments
       PromotionHandler::Cart.new(order, line_item).activate
       Adjustable::AdjustmentsUpdater.update(line_item)
+      TaxRate.adjust(order, [line_item]) if line_item.quantity == 1
```
Comment (JDutil) [inline core/app/models/spree/order_contents.rb]: A quantity of 1 doesn't necessarily mean line item is being created... Not sure how we can best determine that, but I think we have to pass that around in the options hash.

Comment (JDutil) [inline core/app/models/spree/order_contents.rb]: Perhaps change #add to:
```
     def add(variant, quantity = 1, options = {})
       timestamp = Time.now
       line_item = add_to_line_item(variant, quantity, options)
       options[:line_item_created] = true if timestamp <= line_item.created_at
       after_add_or_remove(line_item, options)
     end
```

Comment (vrosancic) [inline core/app/models/spree/order_contents.rb]: It is not so clean, but it will do the trick, at least as temporarily fix. I will update pull request.

**Comment by vrosancic** [comment]
@JDutil I have made those changes from your comment.

**Comment by JDutil** [comment]
Thanks

**Comment by vrosancic** [comment]
Hi @justindrees,

Yes, it is merged in master. We made some changes after, based on this discussion: https://github.com/spree/spree/issues/6548#event-352487674

Here is the commit: https://github.com/vrosancic/spree/commit/69fa9b9661bcfa953d8592f435e76acf58af12b2

Also, I didn't removed that line 'if quantity_changed?' in my first commit, so everything is merged like it should be.

**Pull request closed without merging**